jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/HttpChannelState.java

```diff
@@ -838,10 +838,12 @@ protected boolean onError(Throwable th)
                 {
                     if (_asyncListeners == null || _asyncListeners.isEmpty())
                     {
-                        if (committed)
-                            return true;
-                        sendError(th);
-                        return false;
+//                        Case 3
+                        return committed;
+//                        if (committed)
+//                            return true;
+//                        sendError(th);
+//                        return false;
```

Comment:
There is nothing explicit in the servlet spec saying that calling `startAsync()` should change the way exceptions thrown from a service method are handled.   However, since the servlet spec allows exceptions to be matched to error pages, which are served with an ERROR dispatch, it would be nonsensical to have a synchronous ERROR dispatch happening whilst an async cycle was in progress.   So if we are ASYNC, we should definitely not call sendError - EVEN IF there are no listeners registered.

So I think this change is good.